src/lib/action/active.ts

```diff
@@ -1,49 +1,58 @@
-    path?: string;
+	className?: string;
+	includeDescendants?: boolean;
+	path?: string;
 }
 
 /**
  * Adds a class to an element when the URL matches a path.
  * If used on a link it extracts the value of `href` by default.
  * Usage: `<a href="/docs" use:active />`
  */
-export const active = (node: HTMLElement, options?: UseActiveOptions): ActionReturn<UseActiveOptions> => {
-    let {
-        className = 'active',
-        includeDescendants = false,
-        path = node.getAttribute('href') ?? '/'
-    } = options ?? {};
+export const active = (
+	node: HTMLElement,
+	options?: UseActiveOptions
+): ActionReturn<UseActiveOptions> => {
+	let {
+		className = 'active',
+		includeDescendants = false,
+		path = node.getAttribute('href') ?? '/'
+	} = options ?? {};
 
-    className = className === '' ? 'active' : className
+	className = className === '' ? 'active' : className;
 
-    const addClass = (pathName: string) => {
-        if (includeDescendants ? pathName === path || pathName.includes(path + '/') : pathName === path) {
-            node.classList.add(className);
-        } else {
-            node.classList.contains(className) && node.classList.remove(className);
-        }
-    }
+	const addClass = (pathName: string) => {
+		if (
+			includeDescendants ? pathName === path || pathName.includes(path + '/') : pathName === path
+		) {
+			node.classList.add(className);
+		} else {
+			node.classList.contains(className) && node.classList.remove(className);
+		}
+	};
 
-    page.subscribe($page => {
-        const pathName = $page.url.pathname;
-        addClass(pathName);
-    })
+	// A derived store will only run the subscribe method if the derived value changes,
+	// so we avoid unecessary addClass() calls where the pathName was not changed
+	const pathNameStore = derived(page, ($page) => $page.url.pathname);
+	const unsubscribeFromPathNameStore = pathNameStore.subscribe(addClass);
 
-    return {
-        update: (options: UseActiveOptions) => {
-            node.classList.contains(className) && node.classList.remove(className);
-            className = (options.className === '') ? 'active' : options.className ?? 'active';
-            includeDescendants = options.includeDescendants ?? false;
-            path = options.path ?? node.getAttribute('href') ?? '/';
-            const pathName = get(page).url.pathname;
-            addClass(pathName);
-        },
-        destroy: () => (node.classList.remove(className))
-    }
-}
\ No newline at end of file
+	return {
+		update: (options: UseActiveOptions) => {
+			node.classList.contains(className) && node.classList.remove(className);
+			className = options.className === '' ? 'active' : options.className ?? 'active';
+			includeDescendants = options.includeDescendants ?? false;
+			path = options.path ?? node.getAttribute('href') ?? '/';
+			// i think this might be redundant, because the we could reuse the subscription above
+			const pathName = get(pathNameStore);
```

Comment:
This is here for when we have a variable className and it's changed on the fly. The old class is removed in line 46 and the new one gets added here. The subscription above is derived from the page store so only changes on path change. We could either change the store to also be reactive to the className var or leave it so.

Reply:
Can we be sure that the className will always have changed on the update callback? Isn't it possible that only the "includeDescendants" was changed, making it redundant to perform all of the other updates?

Reply:
You're right. We compare the new className with the stored value and update it only when needed.

Reply:
I think if we change the update to this this pr is good to go:
```
update: (options: UseActiveOptions) => {
    // Update the class only if the className is different than before, taking into account that undefined and '' result in 'active'.
    if (className !== options.className && !((options.className === '' || options.className === undefined) && className === 'active')) {
        node.classList.contains(className) && node.classList.remove(className);
        className = options.className === '' ? 'active' : options.className ?? 'active';
        const pathName = get(pathNameStore);
        addClass(pathName);
    }
    includeDescendants = options.includeDescendants ?? false;
    path = options.path ?? node.getAttribute('href') ?? '/';			
}
```